client/lib/analytics/impact-affiliate.js

```diff
@@ -0,0 +1,25 @@
+import cookie from 'cookie';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import urlParseAmpCompatible from './utils/url-parse-amp-compatible';
+
+/**
+ * Save Impact Affiliate Click ID in a cookie if it is set as a query parameter.
+ */
+export default function saveImpactAffiliateClickId() {
+	// Read click ID query argument, return early if there is none.
+	const parsedUrl = urlParseAmpCompatible( window.location.href );
+	const clickId = parsedUrl?.searchParams.get( 'irclickid' );
+
+	if ( ! clickId ) {
+		return;
+	}
+
+	document.cookie = cookie.serialize( 'irclickid', clickId, {
+		path: '/',
+		expires: new Date( Date.now() + 30 * 24 * 60 * 60 * 1000 ), // 30 days
```

Comment:
Is there any specific reason why we are setting the cookie to expire in 30 days? I looked through project issues but could not find the related discussion

Reply:
That is because currently, the referral window is 30 days so conversions that happen after that are not going to be rewarded.

Reply:
Makes sense!